pkg/asset/machines/aws/machines.go

```diff
@@ -72,10 +77,70 @@ func Machines(clusterID string, region string, subnets map[string]string, pool *
 			},
 		}
 
+		replicas := int32(total / numOfAZs)
+		if int64(idx) < total%numOfAZs {
+			replicas++
+		}
+		machineSet := machinev1.ControlPlaneMachineSet{
+			TypeMeta: metav1.TypeMeta{
+				APIVersion: "machine.openshift.io/v1",
+				Kind:       "ControlPlaneMachineSet",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "openshift-machine-api",
+				Name:      name,
+				Labels: map[string]string{
+					"machine.openshift.io/cluster-api-cluster": clusterID,
+				},
+			},
+			Spec: machinev1.ControlPlaneMachineSetSpec{
+				Replicas: &replicas,
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"machine.openshift.io/cluster-api-machineset": name,
+						"machine.openshift.io/cluster-api-cluster":    clusterID,
+					},
+				},
+				Template: machinev1.ControlPlaneMachineSetTemplate{
+					MachineType: machinev1.OpenShiftMachineV1Beta1MachineType,
+					OpenShiftMachineV1Beta1Machine: &machinev1.OpenShiftMachineV1Beta1MachineTemplate{
+						FailureDomains: machinev1.FailureDomains{
+							Platform: v1.AWSPlatformType,
+							AWS: &[]machinev1.AWSFailureDomain{
+								{
+									Subnet: &machinev1.AWSResourceReference{
+										Type: machinev1.AWSIDReferenceType,
+										ID:   &subnet,
+									},
+									Placement: machinev1.AWSFailureDomainPlacement{
+										AvailabilityZone: zone,
+									},
+								},
+							},
+						},
+						ObjectMeta: machinev1.ControlPlaneMachineSetTemplateObjectMeta{
+							Labels: map[string]string{
+								"machine.openshift.io/cluster-api-machineset":   name,
+								"machine.openshift.io/cluster-api-cluster":      clusterID,
+								"machine.openshift.io/cluster-api-machine-role": role,
+								"machine.openshift.io/cluster-api-machine-type": role,
+							},
+						},
+						Spec: machineapi.MachineSpec{
+							ProviderSpec: machineapi.ProviderSpec{
+								Value: &runtime.RawExtension{Object: provider},
```

Comment:
You need to remove the `subnet` and `placement.availabilityZone` from the `provider` before you inject it into the CPMS